ngraph/test/onnx/onnx_editor.cpp

```diff
@@ -0,0 +1,145 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <algorithm>
+#include "gtest/gtest.h"
+
+#include "ngraph/file_util.hpp"
+#include "ngraph/op/util/op_types.hpp"
+#include "onnx_import/default_opset.hpp"
+#include "onnx_import/editor/editor.hpp"
+#include "onnx_import/onnx.hpp"
+#include "util/test_control.hpp"
+
+NGRAPH_SUPPRESS_DEPRECATED_START
+
+using namespace ngraph;
+
+static std::string s_manifest = "${MANIFEST}";
+
+namespace
+{
+    using InputTypePred = std::function<bool(const std::shared_ptr<ngraph::Node>)>;
+
+    // A higher order factory function that produces predicates bound to a particular element type
+    InputTypePred element_type_is(const element::Type et)
+    {
+        return [et](const std::shared_ptr<ngraph::Node> input) {
+            return std::dynamic_pointer_cast<onnx_import::default_opset::Parameter>(input)
```

Comment:
I am not sure why this dynamic to `Parameter` is needed, as I remember `get_element_type` is defined in `ngraph::Node`